Assets/Scripts/Entity/EntityHealth.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Mirror;
+using UnityEngine.Events;
+
+public class EntityHealth : NetworkBehaviour
+{
+    [SyncVar]
+    private float health = 30f;
+    public float Health => health;
+
+    public UnityEvent OnDie;
+    public UnityEvent OnTakeDamage;
```

Comment:
Prefer C# events unless for UI changes? (I also need to look into the ScriptableObject events we talked about)

Reply:
I'm leaving these as `UnityEvent` because we're most likely going to add listeners for sounds effects and visuals that is going to be part of the same prefab

Reply:
> Prefer C# events unless for UI changes? (I also need to look into the ScriptableObject events we talked about)

I'm leaving these as UnityEvent because we're most likely going to add listeners for sounds effects and visuals that is going to be part of the same prefab